src/background/services/wallet/WalletService.ts

```diff
@@ -229,7 +233,23 @@ export class WalletService implements OnUnlock {
     if (secretType === SecretType.Seedless) {
       const accountIndexToUse =
         accountIndex === undefined ? secrets.account.index : accountIndex;
-      const addressPublicKey = secrets.publicKeys[accountIndexToUse];
+
+      const derivationPaths = await this.addressResolver.getDerivationPathsByVM(
+        accountIndexToUse,
+        secrets.derivationPathSpec,
+        [NetworkVMType.AVM, NetworkVMType.EVM],
+      );
+
+      const derivationPath =
+        isXchainNetwork(network) || isPchainNetwork(network)
+          ? derivationPaths[NetworkVMType.AVM]
+          : derivationPaths[NetworkVMType.EVM];
+
+      const addressPublicKey = getPublicKeyFor(
+        secrets,
+        derivationPath,
+        'secp256k1',
+      );
```

Comment:
I'll make it smarter when I implement Seedless signing for Solana. This just makes it work for the next week or so.